substrate/frame/staking-async/src/pallet/mod.rs

```diff
@@ -1319,6 +1354,88 @@ pub mod pallet {
 				T::DbWeight::get().reads(1)
 			}
 		}
+
+		/// Execute one step of era pruning and get actual weight used
+		fn do_prune_era_step(era: EraIndex) -> Result<Weight, DispatchError> {
+			// Get current pruning state. If EraPruningState doesn't exist, it means:
+			// - Era was never marked for pruning, OR
+			// - Era was already fully pruned (pruning state was removed on final step)
+			// In either case, this is an error - user should not call prune on non-prunable eras
+			let Some(current_step) = EraPruningState::<T>::get(era) else {
+				return Err(Error::<T>::EraNotPrunable.into());
+			};
+
+			// Simple item-based limiting - no complex weight calculations
+			let items_limit = T::MaxPruningItems::get();
+
+			let actual_weight = match current_step {
+				PruningStep::ErasStakersPaged => {
+					let result = ErasStakersPaged::<T>::clear_prefix((era,), items_limit, None);
+					let items_deleted = result.backend as u32;
+					// Update pruning state if this step is complete
+					if result.maybe_cursor.is_none() {
+						EraPruningState::<T>::insert(era, PruningStep::ErasStakersOverview);
+					}
+					// Return benchmark weight for actual items processed
+					T::WeightInfo::prune_era_stakers_paged(
+						items_deleted.min(T::MaxValidatorSet::get()),
```

Comment:
why min with `MaxValidatorSet`?

Reply:
This ensures we never account for deletion of more than `MaxValidatorSet` items. If for some reason (`MaxValidatorSet` was decreased) more items got deleted than validator set, we should still try to account for that weight, right?

Or we want to limit how many we actually delete, we should

`let items_limit = T::MaxPruningItems::get().min(T::MaxValidatorSet::get());`